### PR TITLE
feat: add MongoDB database as a part of the CLI

### PR DESCRIPTION
**Pull request by pythonhubdev**
## Overview

Fixed MongoDB test failures in generated FastAPI projects. The test files were hardcoded for SQLAlchemy but failed when MongoDB was selected as the database type. Made all test-related templates conditional based on `database_type` to support both SQLAlchemy and MongoDB configurations.

## Type of Change

- [x] Bug fix

## Scope / Affected Areas
- [x] Developer tooling

## Breaking Changes

- [x] No

## Public API Impact (Python SDK)

- New modules/classes/functions:
- Changed/removed APIs:
- Deprecations (timeline and migration path):

## Testing & Verification

- Commands run locally:
	- [x] task test (pytest)
	- [x] task lint (ruff/pyright/mypy)
	- [x] pre-commit run --all-files (optional)
- Added/updated tests: [ ] unit [ ] integration
- Key scenarios and edge cases:
  - Generated FastAPI projects with MongoDB database type now have working tests
  - SQLAlchemy projects continue to work as before
  - Both database types use appropriate connection patterns and fixtures

## Documentation
- Notes:
  - MongoDB support in scaffoldr now includes proper test configuration

## Environment / Config Changes

- New/changed env vars:
- Defaults and validation:
- Backwards compatible: [yes]

## Checklist

- [x] Code formatted and linted (ruff format/check)
- [x] Type checks pass (pyright, mypy)
- [x] No secrets/keys committed

### CODE REVIEW

File: templates/fastapi_template/{{ project_name }}/.env.jinja [hunk 1]
```diff
@@ -13,8 +13,15 @@
 {{ project_slug|upper }}_LOG_GRANIAN_ERROR_LEVEL=40
 
 {% if database %}
+{% if database_ype == 'postgresql' %}
```
[the remark this answers is not in the document]
Comment (pythonhubdev) [reply templates/fastapi_template/{{ project_name }}/.env.jinja:16]: Resolved

File: templates/fastapi_template/{{ project_name }}/tests/test_health.py.jinja [hunk 2]
```diff
@@ -30,4 +33,17 @@ class TestHealth:
 		assert "status" in data
 		assert data["status"] == "success"
 		# Note: Database health check not implemented in basic health endpoint
+	{% elif database_type == 'mongodb' %}
+	@pytest.mark.asyncio
+	async def test_health_endpoint_with_mongodb(
+		self, client: TestClient, mongodb_client: None,
+	) -> None:
```
[the remark this answers is not in the document]
Comment (pythonhubdev) [reply templates/fastapi_template/{{ project_name }}/tests/test_health.py.jinja:40]: Resolved
